test-e2e/cip/push-golden.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# About: This script automatest the process of pushing golden iamges for e2e tests.
+# Images are loaded from local archives and pushed to the designated staging repo.
+# The script is triggered within e2e.go during test setup.
+#
+# Usage:
+#   ./push-golden.sh repo-root
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if (( $# != 1 )); then
+    echo -e "Error: Invalid number of arguments!
+    Usage: publish-golden.sh repo-root"
+    exit 1
+fi
+
+repoRoot=$1
+archivePath="${repoRoot}/test-e2e/cip/golden-archives"
+eval "$(~/k8s-container-image-promoter/workspace_info.sh)"
```

Comment:
You should use `source`or `.` instead of `eval`.

Reply:
> You should use `source`or `.` instead of `eval`.

On second thought, for consistency sake let's just use `eval`. See my comment below about using/not-using `cat`.

Reply:
Thinking about this again, probably `source <(${repoRoot}/workspace_info.sh)` is the best way. The `<(..)` syntax is called process substitution.

Reply:
Changing. Out of curiosity, why might this be the _better_ way to load workspace variables?

Reply:
See https://stackoverflow.com/questions/17529220/why-should-eval-be-avoided-in-bash-and-what-should-i-use-instead